rtsp.go

```diff
@@ -164,47 +203,52 @@ func (rc *rtspCamera) reconnectClient() (err error) {
 		return err
 	}
 
-	// setup RTP/H264 -> H264 decoder
 	rtpDec, err := format.CreateDecoder2()
 	if err != nil {
-		return err
+		rc.logger.Errorf("error creating H264 decoder %v", err)
 	}
 
-	// setup H264 -> raw frames decoder
-	rc.rawDecoder, err = newH264Decoder()
+	rc.rawDecoder, err = newDecoder(C.AV_CODEC_ID_H264)
 	if err != nil {
 		return err
 	}
 
-	// if SPS and PPS are present into the SDP, send them to the decoder
+	// For H.264, handle SPS and PPS (assuming your H264 format has these fields)
 	if format.SPS != nil {
 		rc.rawDecoder.decode(format.SPS)
 	}
 	if format.PPS != nil {
 		rc.rawDecoder.decode(format.PPS)
 	}
 
+	iFrameReceived := false
+
 	// On packet retreival, turn it into an image, and store it in shared memory
 	rc.client.OnPacketRTP(track, format, func(pkt *rtp.Packet) {
 		// extract access units from RTP packets
-		au, _, err := rtpDec.Decode(pkt)
+		au, _, err := rtpDec.DecodeUntilMarker(pkt)
 		if err != nil {
 			if err != rtph264.ErrNonStartingPacketAndNoPrevious && err != rtph264.ErrMorePacketsNeeded {
 				rc.logger.Errorf("error decoding(1) h264 rstp stream %v", err)
 			}
 			return
 		}
 
-		for _, nalu := range au {
+		// wait for an I-frame
+		if !iFrameReceived {
+			if !h264.IDRPresent(au) {
+				rc.logger.Warnf("waiting for I-frame")
+				return
+			}
+			iFrameReceived = true
+		}
 
+		for _, nalu := range au {
 			if len(nalu) < 20 {
-				// TODO(ERH): this is probably wrong, but fixes a spam issue with "no frame!"
+				rc.logger.Warnf("nalu too short")
```

Comment:
More context in this warnign statement - what can a user do here, or expect from camera behaviour if the nalu is too short?

Reply:
- Still digging into why the h264 decoder is getting these short nalus. Do not have this problem on h265.
- Added back in `TODO` comment.
- Tested with and without this nalu length checker and they both work.